opentelemetry/proto/profiles/v1development/profiles.proto

```diff
@@ -502,3 +499,12 @@ message Function {
   // Line number in source file. 0 means unset.
   int64 start_line = 4;
 }
+
+// A custom 'dictionary native' style of encoding attributes which is more specialized
+// for our use than opentelemetry.proto.common.v1.KeyValue
+// Specifically, uses the string table for keys and allows optional unit information.
+message KeyValueAndUnit {
+  int32 key_strindex  = 1;
+  opentelemetry.proto.common.v1.AnyValue value = 2;
+  int32 unit_strindex = 3; // non-zero where the unit cannot be derived from the name by semconv.
```

Comment:
```suggestion
  int32 unit_strindex = 3; // non-zero if the unit is not defined by semconv.
```

Reply:
We should make it clear here that not all keys will have units as otherwise this reads a bit strange.